internal/clients/teamSOSA/modules/AgentParameters.go

```diff
@@ -74,6 +76,24 @@ func (ap *AgentParameters) UpdateTrustValue(agentID uuid.UUID, eventValue float6
 	ap.TrustNetwork[agentID] = clamp(ap.TrustNetwork[agentID])
 }
 
+// updates: the agents regime trust values
+func (ap *AgentParameters) UpdateRegimeTrustValues(rankOrder []utils.Governance) {
+	for rank, governance := range rankOrder {
+		if _, ok := ap.RegimeTrust[governance]; !ok {
+			// if governance is not in our trust network,i.e. at the start, assign starting value of 0.5
+			ap.RegimeTrust[governance] = 0.5
+		} else {
+			if rank == 0 {
+				ap.RegimeTrust[governance] += 0.05
+				ap.RegimeTrust[governance] = clamp(ap.RegimeTrust[governance])
+			} else if rank == 2 {
+				ap.RegimeTrust[governance] -= 0.05
+				ap.RegimeTrust[governance] = clamp(ap.RegimeTrust[governance])
```

Comment:
Ok, this code makes my previous two comments irrelevant. Did we not decide that the regime score is just the proportion of total resources that they brought in?

Reply:
Unless this is trying to make the rankings more 'definitive', by shifting best and worst a bit? If so, this can still be handled globally, right?

Reply:
Because using proportion makes sure that it's constrained within [0, 1]. If you wanted the dynamic-ness, I'd attach a method to the server, and inject it into the agent (Inversion of Control). The agent can then invoke this method on the server (which itself is dynamically changing)